packages/core/datadog-sourcemaps.gradle

```diff
@@ -0,0 +1,139 @@
+import org.apache.tools.ant.taskdefs.condition.Os
+
+def reactConfig = project.hasProperty("react") ? project.react : [:]
+def reactRoot = file(reactConfig.root ?: "../../")
+
+def datadogConfig = new Properties()
+def datadogConfigFile = new File(reactRoot, "datadog-sourcemaps.properties")
+
+if (datadogConfigFile.exists()) {
+    datadogConfigFile.withInputStream {
+        datadogConfig.load(it)
+    }
+}
+
+afterEvaluate {
+
+    def androidExtension = extensions.findByName("android")
+
+    if (androidExtension == null) {
+        throw new GradleException("Android extension is missing.")
+    }
+
+    // TODO service name per variant?
+    def serviceName = datadogConfig.SERVICE_NAME ?: androidExtension.defaultConfig.applicationId
+
+    if (serviceName == null) {
+        throw new GradleException("Cannot determine service name.")
+    }
+
+    def releaseVersion = androidExtension.defaultConfig.versionName
+    def buildVersion = androidExtension.defaultConfig.versionCode
+
+    if (releaseVersion == null) {
+        throw new GradleException("Cannot determine application version," +
+                " probably not running in the app module.")
+    }
+
+    androidExtension.applicationVariants.all { variant ->
+        // locations and names are defined in "../../node_modules/react-native/react.gradle"
+        def targetName = variant.name.capitalize()
+        def targetPath = variant.dirName
+
+        def bundleTask = tasks.findByName("bundle${targetName}JsAndAssets")
+
+        if (bundleTask == null) {
+            logger.info("Cannot find JS bundle task for target=${targetName}.")
+            return
+        }
+        if (!bundleTask.enabled) {
+            logger.info("JS bundle task for target=${targetName} is not enabled.")
+            return
+        }
+
+        def bundleAssetName = reactConfig.bundleAssetName ?: "index.android.bundle"
+
+        def jsBundleDir = file("$buildDir/generated/assets/react/${targetPath}")
+        def jsBundleFile = file("$jsBundleDir/$bundleAssetName")
+
+        def jsSourceMapsDir = file("$buildDir/generated/sourcemaps/react/${targetPath}")
```

Comment:
Question: does this make us rely too much on the RN react.gradle implementation?

Reply:
we could get the --sourcemap-output argument from the task instead. That is public API so less likely to change

Reply:
As discussed, let's go for this implementation as we would rely on the fact that the command uses the bundle tool anyway with the other solution.